sensor/extrinsic_reflector_based_calibrator/src/extrinsic_reflector_based_calibrator.cpp

```diff
@@ -1280,21 +1336,169 @@ void ExtrinsicReflectorBasedCalibrator::calibrateSensors()
       "The calibrated poses differ considerably with the initial calibration. This may be either a "
       "fault of the algorithm or a bad calibration initialization");
   }
+
+  output_metrics_.push_back(static_cast<float>(converged_tracks_.size()));
+  output_metrics_.push_back(static_cast<float>(calibrated_2d_distance_error));
+  output_metrics_.push_back(static_cast<float>(calibrated_2d_yaw_error));
+}
+
+void ExtrinsicReflectorBasedCalibrator::findCombinations(
+  int n, int k, std::vector<int> & curr, int first_num,
+  std::vector<std::vector<int>> & combinations)
+{
+  int curr_size = static_cast<int>(curr.size());
+  if (curr_size == k) {
+    combinations.push_back(curr);
+    return;
+  }
+
+  int need = k - curr_size;
+  int remain = n - first_num + 1;
+  int available = remain - need;
+
+  for (int num = first_num; num <= first_num + available; num++) {
+    curr.push_back(num);
+    findCombinations(n, k, curr, num + 1, combinations);
```

Comment:
This time is not an issue since calibration tools can be slow (no action is required this time).
However, remember to avoid as much as possible using recursive solutions.
An alternative is to use stacks for example
An additional improvement would be to reserve the size of the vector since we know the number of combinations